frontend/src/app/services/session.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
+import { Observable, BehaviorSubject } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+const URL = 'http://localhost:8080/';
+const httpOptions = {
+  headers: new HttpHeaders({
+    'Authorization': 'Basic ' + btoa("fooClientIdPassword:secret"),
+    'Content-type': 'application/x-www-form-urlencoded'
+  })
+};
+
+@Injectable()
+
+export class SessionService {
+
+  private _userData$: BehaviorSubject<any> = new BehaviorSubject(undefined);
```

Comment:
change any to valid User object or flag this line as TODO, because there will be a User type later on
```suggestion
  // TODO: change any to valid User type
  private _userData$: BehaviorSubject<any> = new BehaviorSubject(undefined);
```